json_theme/lib/src/codec/theme_decoder.dart

```diff
@@ -3583,10 +3582,13 @@ class ThemeDecoder {
   ///  * [decodeEdgeInsetsGeometry]
   ///  * [decodeShapeBorder]
   ///  * [decodeTextStyle]
-  static DialogTheme? decodeDialogTheme(dynamic value, {bool validate = true}) {
-    DialogTheme? result;
+  static DialogThemeData? decodeDialogTheme(
```

Comment:
This would need to become `decodeDialogThemeData` or else downstream code won't be able to properly detect it.